replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/modern/ReplicationPersistentStoreImpl.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.modern;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ditto.replication.api.modern.Persistable;
+import org.codice.ditto.replication.api.modern.ReplicationPersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReplicationPersistentStoreImpl implements ReplicationPersistentStore {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(ReplicationPersistentStoreImpl.class);
+
+  private static final int DEFAULT_PAGE_SIZE = 100;
+
+  private static final int DEFAULT_START_INDEX = 0;
+
+  private PersistentStore persistentStore;
+
+  public ReplicationPersistentStoreImpl(PersistentStore persistentStore) {
+    this.persistentStore = persistentStore;
+  }
+
+  @Override
+  public <T extends Persistable> Optional<T> get(Class<T> klass, String typeName, String id) {
+    String cql = String.format("'id' = '%s'", id);
+    List<Map<String, Object>> storedMaps = Collections.emptyList();
+    try {
+      storedMaps = persistentStore.get(typeName, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+    } catch (PersistenceException e) {
+      LOGGER.debug(
+          "An error occurred while retrieving a {} with an ID of {}", klass.getName(), id, e);
```

Comment:
✏️ This would result in leaving the map empty and the next debug trace to come out before returning `Optional.empty()`. Should we just return here? That being said, I am still wondering why we are not propagating the error wrapped into a new one specific to our API to the client's of this method to let them know you are unable to retrieve anything as opposed to telling them there is nothing for that id.